Feat: Implement "Delete account" UI

kanta1207 opened the pull request:
<!--

Title Naming Conventions:

- Start with a capitalized prefix followed by an imperative form of a verb.
- Refer to the "Types of Branches" section for the prefix:
  https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches

Example:
- Feature: Add button to the home page
- Fix: Resolve error in data fetching
- Hotfix: Fix user authentication issue
- Refactor: Reorganize the file structure
- Test: Add unit tests for form validation
- Docs: Update installation guide

-->

## Overview

<!-- Provide a brief description of the changes introduced by this PR. -->

Implement the UI of "Delete account" feature in Profile page.

## Changes

<!-- List the changes -->
<!-- Delete this section if not needed -->

- Implement dropdown menu in profile page header
- Implement `DeleteAccountDialog` in profile page 
- Add 16px align offset to the dropdown menu in group single page header (Out of the issue scope)
- Implement `LogoutDialog` in profile page (Out of the issue scope) 


## Screen Captures

<!-- Add screenshots or screen recordings of the changes -->
<!-- Necessary for UI changes -->
<!-- Delete this section if not needed -->
### `LogoutDialog` and `DeleteAccountDialog` in Profile page

https://github.com/nishiki-tech/nishiki-frontend/assets/99339182/523d6e7a-7907-470e-98c9-12750597687b

### 16px align offset of DropdownMenu in GroupSinglePage header
<img width="406" alt="Screenshot 2024-03-09 at 7 48 25" src="https://github.com/nishiki-tech/nishiki-frontend/assets/99339182/28267b48-a96c-43fa-a2ae-592a5e558b2e">


## Assignee Checklist:

<!-- You can tick the checkboxes if not applicable -->

- [x] The naming convention of the PR title is correct (See the comment at the top of this template)
- [x] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [x] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [x] The correct assignees and reviewers have been designated for this PR
- [x] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] All the related issues are associated with this PR
- [x] All criteria in the associated issues are met (please tick the checkboxes)
- [x] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

<!-- You can tick the checkboxes if not applicable -->

- [ ] Double-check the "Assignee Checklist"
- [ ] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] The code is well-commented and easy to understand
- [ ] The UI changes accurately reflect the provided design

kanta1207 commented on src/components/ui/DropdownMenu.tsx:
```diff
@@ -9,7 +9,7 @@
  * "DropdownMenuRadioGroup"
  * See the official docs for more info:
  * shadcn/ui: https://ui.shadcn.com/docs/components/dropdown-menu
```
Changed the reference link to correct one

kanta1207 commented on src/components/ui/Checkbox.tsx:
Changed the reference link to correct one

kanta1207 commented on src/components/ui/Button.tsx:
```diff
@@ -18,6 +18,8 @@ const buttonVariants = cva(
         none: '',
         primary: 'bg-primary text-white enabled:hover:bg-primary-dark disabled:opacity-50',
         error: 'bg-danger text-white enabled:hover:bg-danger-dark disabled:opacity-50',
```
Does naming `danger-outline` make sense to you? Please share with me if you have any better ideas

nick-y-ito replied:
I suggest aligning the red-filled and red-outlined buttons by using either words 'danger' or 'error'. 

I suggest naming them `danger` and `danger-outline` because, in our Figma design, it's called "danger". I think they were initially named with the word `error`, and that's why we named the color variant `error`, by the way.

<img width="400" alt="image" src="https://github.com/nishiki-tech/nishiki-frontend/assets/99148565/95594977-3c6d-4e6d-bef2-a0ee9713698e">

<img width="400" alt="image" src="https://github.com/nishiki-tech/nishiki-frontend/assets/99148565/bdc8a0df-3302-41ac-98e0-54b504d0112d">

nick-y-ito replied:
> I suggest naming them danger and danger-outline 

@kanta1207

What I meant here was to name them as `danger` and `danger-outline` instead of `error` and `error-outline`.

kanta1207 commented on src/features/groups/components/GroupSingleHeaderDropdownMenuTriggerButton/GroupSingleHeaderDropdownMenuContent/index.tsx:
Added align offset for GroupSingle page header dropdown

nick-y-ito commented on src/features/profile/components/ProfileHeaderDropdownMenuTriggerButton/ProfileHeaderDropdownMenuContent/DeleteAccountDialogTriggerButton/DeleteAccountDialogContent.tsx:
Though this is an edge case, the alignment may need to be adjusted because it is not very nice when the text has a line break.

<img width="380" alt="image" src="https://github.com/nishiki-tech/nishiki-frontend/assets/99148565/0b5ed0ce-02d1-45c3-a7b7-c13e89eabbec">

kanta1207 replied:
<img width="508" alt="Screenshot 2024-03-09 at 23 26 55" src="https://github.com/nishiki-tech/nishiki-frontend/assets/99339182/d6d0b49a-ce79-4d2b-929b-6876d530ca2e">

> Though this is an edge case, the alignment may need to be adjusted because it is not very nice when the text has a line break.

Thank you for pointing out! 
After fixing the side padding to `px-4` from `px-6`, now the line break will not appear until width of 325px, 
And even with width 320px(which is supposed to be minimum width of mobile device we should care for), 
it looks kinda fine to me. Could you kindly share with me how you want it to look like to be a better UI?

<img width="474" alt="Screenshot 2024-03-09 at 23 32 17" src="https://github.com/nishiki-tech/nishiki-frontend/assets/99339182/999b54f0-5350-4a30-b2e9-9fd95e1eaa1e">

nick-y-ito replied:
As you say, I think the current state is acceptable. It would take only 3 minutes to fix, though. It's no longer a big matter.

nick-y-ito reviewed (changes requested):
@kanta1207 

Thank you for the implementation! Please take a look at the comments that I left inline.

In addition to those comments, I would like to ask you to rename all "log out" to "sign out", although "log in" and "log out" are used in Figma. It is because we cannot rephrase the "Sign in with Google" button as it's injected by a third-party library.

I will leave you to decide whether or not to rename all the existing `log in` to `sign in`, including variables, functions, components, and directory names. This may over-expand this PR's scope.

kanta1207 commented:
@nick-y-ito 
Thank you for the reviews! 
I fixed the points you mentioned, and also left a comment on the points I wasn't so sure about.

About the refactoring, I did the renaming of `Logout` to `SignOut`. 
(I used camel case because `signout` cause an spelling caution in the editor.)
I didn't do much on the directory structure, so `enter` or `space` key doesn't work in the delete or sign out dialog, although it might not be necessary for these two tho.

nick-y-ito commented on src/features/groups/components/GroupSingleHeaderDropdownMenuTriggerButton/GroupSingleHeaderDropdownMenuContent/index.tsx:
```diff
@@ -37,7 +37,7 @@ export const GroupSingleHeaderDropdownMenuContent = ({
 
   return (
-    <DropdownMenuContent>
+    <DropdownMenuContent alignOffset={16}>
```
Could you please add a comment that explains this offset size must match the size of theside padding of the body?

nick-y-ito reviewed (changes requested):
@kanta1207

Thank you for the update! I have left comments and replied to you messages. Please take a look.

Also,

- [x] Could you please replace all the "LogOut" with "SignOut"? I found some still remaining, including `IconLogOut`.

kanta1207 commented:
@nick-y-ito 
Thank you for your suggestions!
I added comments and refactor the variable/variant names following your suggestions, 
please have a look when you have a moment 🙏

nick-y-ito commented on src/assets/images/icons/index.ts:
```diff
@@ -25,3 +24,4 @@ export { default as IconPlus } from './icon_plus.svg';
 export { default as IconProfileOff } from './icon_profile-off.svg';
 export { default as IconProfileOn } from './icon_profile-on.svg';
 export { default as IconSearch } from './icon_search.svg';
+export { default as IconSignOut } from './icon_SignOut.svg';
```
The latter part of the file name should be `kebab-case`.

nick-y-ito reviewed (changes requested):
@kanta1207 
I left additional comments inline. Please take a look!

Also, 

- [x] Please fix the build error.

> - [x] Could you please replace all the "LogOut" with "SignOut"? I found some still remaining, including IconLogOut.

- [x] Could you recheck all the files about renaming "log out" to "sign out", please?

<img width="300" alt="image" src="https://github.com/nishiki-tech/nishiki-frontend/assets/99148565/718aed8e-cb3b-4ff1-ac57-e7988153f03a">

kanta1207 commented:
@nick-y-ito 
> What I meant here was to name them as danger and danger-outline instead of error and error-outline.

Sorry, it's just an excuse but I thought I left a comment like "changing all of `error` variant to `danger` would be too big, so I chose to align it `error` & `error-outline` for now", but it was pending and couldn't deliver to you.  **Just wanted to let you know I didn't mean to ignore your suggestion or anything** 🙏 
And, at the end I aligned all of them into `danger` & `danger-outline`. I just changed my mind.

Also, sorry for missing the build error! I just fixed them. Please take a look when you have a time again, kindly 🙏

nick-y-ito reviewed (changes requested):
@kanta1207 
> Sorry, it's just an excuse but I thought I left a comment like "changing all of error variant to danger would be too big, so I chose to align it error & error-outline for now", but it was pending and couldn't deliver to you. Just wanted to let you know I didn't mean to ignore your suggestion or anything 🙏

Never mind😂 I thought you were just misunderstanding, now I know what you were trying to do. Anyway, thank you for fixing all!

However, you haven't fixed the renaming from "log out" to "sign out". Could you re-read the last comment, please?

kanta1207 commented:
@nick-y-ito 
I renamed the `log out` to `sign out`! Please have a look when you have a moment, again 🙏

nick-y-ito reviewed (approved):
@kanta1207 
Thank you so much for your implementation and refactoring! I appreciate it.

This PR looks absolutely good for merging!

kanta1207 commented:
Thank you for all of your reviews! I'm gonna merge this PR 👍
